docs/user/testing-metrics.md

```diff
@@ -98,11 +98,64 @@ Note that each of these functions is marked as `internal`, you need to import `G
 
 </div>
 
+<div data-lang="Python" class="tab">
+
+It is generally a good practice to "reset" Glean prior to every unit test that uses Glean, to prevent side effects of one unit test impacting others.
+Glean contains a helper function `glean.testing.reset_glean()` for this purpose.
+It has two required arguments: the application id, and the application version.
+Each reset of Glean will create a new temporary directory for Glean to store its data in.
+This temporary directory is automatically cleaned up the next time Glean is reset or when the testing framework finishes.
+
+The instructions below assume using `py.test` as the test runner.
+Other test-running libraries have similar features, but are different in the details.
+
+To have this function run before every unit test in a `py.test` module:
+
+```python
+from glean import testing
+
+def setup_module():
+    testing.reset_glean("my-app-id", "0.1.0")
+```
+
+To have this function run before every test in an entire test suite, add the following to the `conftest.py` file:
+
+```python
+from glean import testing
+
+def pytest_runtest_module():
+    testing.reset_glean("my-app-id", "0.1.0")
+```
```

Comment:
```python
import pytest
import glean


@pytest.fixture(name="reset_glean", auto=True)
def fixture_reset_glean():
    """Reset Glean before every test item."""
    testing.reset_glean("my-app-id", "0.1.0")
```

Reply:
Doesn't using a fixture require specifying the fixture in every test that uses Glean?  e.g.:

```
def test_thing(fixture_reset_glean):
    pass
```

Or is that what the `auto` flag does (which I can't find the docs for)?

Reply:
Sorry, my bad. It's `autouse`.

See https://docs.pytest.org/en/latest/reference.html#pytest-fixture 📝 

I would say the idiomatic way to implemented the described tasks with pytest would be:

```python
import pytest
import glean


@pytest.fixture(name="run_for_every_test", scope="function", autouse=True)
def fixture_run_for_every_test():
    pass

@pytest.fixture(name="run_once_for_session", scope="session", autouse=True)
def fixture_once_for_session():
    pass
```

Reply:
Great.  I'll update the docs here.